### ISSUE TITLE
Hitting "up" does not bring the whole multi-line entry from history.

### ISSUE THREAD

**Issue by kubouch** [nushell/engine-q#436]
Running latest reedline after `cargo update` (a44a590b25e2fb2593db24a17c06ed887633a5fc).

```
> def foo [] { 
    "foo"
}

> }  # <-- appears after hitting 'up'
```

Interestingly, reedline by itself handles this fine. It must be something within engine-q.

**Comment by sholderbach** [comment]
Reedline is currently still not able to store the multiline entries correctly in a file. So as soon as the `Reedline` object is dropped the multiline entries become individual lines 
https://github.com/nushell/reedline/issues/116

**Comment by sholderbach** [comment]
Closed by nushell/reedline#207 and nushell/reedline#210 and was pulled into engine-q with #562
